README.md

```diff
@@ -1,96 +1,119 @@
-
-* User Guides
-  * [README-agent-install.md](https://github.com/cisco/cisco-network-puppet-module/blob/develop/docs/README-agent-install.md) : Agent Installation and Configuration Guide
-  * [README-beaker-agent-install.md](https://github.com/cisco/cisco-network-puppet-module/blob/develop/docs/README-beaker-agent-install.md) : Automated Agent Installation and Configuration
-  * [README-package-provider.md](https://github.com/cisco/cisco-network-puppet-module/blob/develop/docs/README-package-provider.md) : Cisco Nexus Package Management using the Package Provider
-  * [README-example-manifest.md](https://github.com/cisco/cisco-network-puppet-module/blob/develop/examples/README.md) : Example Demo Manifest User Guide
-  * The remainder of this document is aimed at end users
-* Developer Guides
-  * [CONTRIBUTING.md](https://github.com/cisco/cisco-network-puppet-module/blob/develop/CONTRIBUTING.md) : Contribution guidelines
-  * [README-develop-types-providers.md](https://github.com/cisco/cisco-network-puppet-module/blob/develop/docs/README-develop-types-providers.md) : Developing new ciscopuppet Types and Providers
-  * [README-develop-beaker-scripts.md](https://github.com/cisco/cisco-network-puppet-module/blob/develop/docs/README-develop-beaker-scripts.md) : Developing new beaker test scripts for ciscopuppet
-* Maintainers Guides
-  * [README-maintainers.md](https://github.com/cisco/cisco-network-puppet-module/blob/develop/docs/README-maintainers.md) : Guidelines for core maintainers of the ciscopuppet project
-  * All developer guides apply to maintainers as well
-
-Please see [Learning Resources](#learning-resources) for additional references.
-
---
 #### Table of Contents
 
-1. [Overview](#overview)
 1. [Module Description](#module-description)
 1. [Setup](#setup)
-1. [Usage](#usage)
-1. [Platform Support](#platform-support)
-   * [Provider Support Across Platforms](#provider-support-across-platforms)
+1. [Example Manifests](#example-manifests)
 1. [Resource Reference](#resource-reference)
    * [Resource Type Catalog (by Technology)](#resource-by-tech)
    * [Resource Type Catalog (by Name)](#resource-by-name)
-1. [Limitations - OS compatibility, etc.](#limitations)
-1. [Cisco OS Differences](#cisco-os-differences)
-1. [Learning Resources](#learning-resources)
-
-
+   * [Resource Platform Support Matrix](#resource-platform-support-matrix)
+1. [Documentation Guide](#documentation-guide)
 
-## Overview
+## <a href='module-description'>Module Description</a>
 
 The ciscopuppet module allows a network administrator to manage Cisco Network Elements using Puppet. This module bundles a set of Puppet Types, providers, Beaker Tests, Sample Manifests and Installation Tools for effective network management.  The  resources and capabilities provided by this Puppet Module will grow with contributions from Cisco, Puppet Labs and the open source community.

```

Comment:
Can you make `providers` uppercase in the second sentence?  `This module bundles a set of Puppet Types, Providers, Beaker...`